clients/src/main/java/org/apache/kafka/clients/ClientUtils.java

```diff
@@ -107,7 +108,9 @@ public static ChannelBuilder createChannelBuilder(AbstractConfig config, Time ti
 
     static List<InetAddress> resolve(String host, HostResolver hostResolver) throws UnknownHostException {
         InetAddress[] addresses = hostResolver.resolve(host);
-        return filterPreferredAddresses(addresses);
+        List<InetAddress> result= filterPreferredAddresses(addresses);
+        log.debug("Resolved host {} as {}", host, result.stream().map(i -> i.getHostAddress()).collect(Collectors.toList()));
```

Comment:
You are computing the collection even if the log is not enabled.

Reply:
added a guard